lib/maths/CTimeSeriesModel.cc

```diff
@@ -964,17 +966,28 @@ CUnivariateTimeSeriesModel::confidenceInterval(core_t::TTime time,
             {m_IsNonNegative ? std::max(result[2], 0.0) : result[2]}};
 }
 
-bool CUnivariateTimeSeriesModel::forecast(core_t::TTime startTime,
+bool CUnivariateTimeSeriesModel::forecast(core_t::TTime firstDataTime,
+                                          core_t::TTime lastDataTime,
+                                          core_t::TTime startTime,
                                           core_t::TTime endTime,
                                           double confidenceInterval,
                                           const TDouble2Vec& minimum_,
                                           const TDouble2Vec& maximum_,
                                           const TForecastPushDatapointFunc& forecastPushDataPointFunc,
                                           std::string& messageOut) {
-    if (m_ResidualModel->isNonInformative()) {
+
+    core_t::TTime horizon{std::min(lastDataTime + (lastDataTime - firstDataTime),
+                                   lastDataTime + m_TrendModel->maximumForecastInterval())};
```

Comment:
I think we should still have a sensible upper limit just to prevent someone requesting a `1000 years forecast` although he might need to tweak the data set to do that.

Reply:
This is ensured by `m_TrendModel->maximumForecastInterval()`. For our current model parameterisation the limit works out at just over 1600 days. I think I'd prefer to tie this to the model behaviour rather than use a fixed hard limit. I'll add a comment to this effect.